lib/core/include/irods/base64.h

```diff
@@ -1,27 +1,26 @@
-/*** Copyright (c), The Regents of the University of California            ***
- *** For more information please refer to files in the COPYRIGHT directory ***/
+#ifndef IRODS_BASE64_H
+#define IRODS_BASE64_H
 
-/* base64.h - header file for base64.c
- */
-
-
-
-#ifndef BASE64_H__
-#define BASE64_H__
+#warning "base64.h is deprecated and will be removed in a future iRODS release.  Please use base64.hpp."
 
 #include "irods/rodsDef.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-int base64_encode( const unsigned char *in,  unsigned long inlen,
-                   unsigned char *out, unsigned long *outlen );
-int base64_decode( const unsigned char *in,  unsigned long inlen,
-                   unsigned char *out, unsigned long *outlen );
+// clang-format off
+
+__attribute__((deprecated("base64_encode is deprecated. Use irods::base64_encode instead.")))
+int base64_encode(const unsigned char* in, unsigned long inlen, unsigned char* out, unsigned long* outlen);
+
+__attribute__((deprecated("base64_decode is deprecated. Use irods::base64_decode instead.")))
+int base64_decode(const unsigned char* in, unsigned long inlen, unsigned char* out, unsigned long* outlen);
```

Comment:
From my testing it seems to work fine using these in a test program.  You just get the warning on compile.  After removing the warning on the header itself, iRODS compiles fine as it isn't actually calling these methods.

Reply:
As we discovered previously, there's overlap with declarations from another library. I think it would be preferable to deprecate the header itself rather than the declarations, but I don't feel super strongly about it.

Reply:
The header has been deprecated and the declarations left alone.